#! /bin/sh /usr/share/dpatch/dpatch-run
## -*- diff -*-
## real-no-history.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## DP: Only save history if Vsaving_history is true.  Thanks to John Eaton
## DP: for the patch.  Closes Bug#514802.

@DPATCH@
--- octave3.0-3.0.4~rc3.orig/src/toplev.cc
+++ octave3.0-3.0.4~rc3/src/toplev.cc
@@ -643,7 +643,8 @@
 
       octave_history_write_timestamp ();
 
-      command_history::clean_up_and_save ();
+      if (Vsaving_history)
+	command_history::clean_up_and_save ();
 
       close_files ();
 
